youtube_dl/extractor/markiza.py

```diff
@@ -26,54 +28,89 @@ class MarkizaIE(InfoExtractor):
         'only_matching': True,
     }, {
-        'url': 'http://videoarchiv.markiza.sk/embed/85295',
+        'url': 'https://videoarchiv.markiza.sk/embed/85295',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        embed = self._search_regex(
+            r'<iframe src="(https:\/\/media.*?)" style="',
+            webpage, 'embed', fatal=False)
+        webpage = self._download_webpage(embed, video_id)
+        data = self._search_regex(
+            r'processAdTagModifier\((\{.*)\), {"video":',
+            webpage, 'embed', fatal=False)
+        data2 = self._search_regex(
+            r'processAdTagModifier\(\{.*\), ({"video":.*)\);',
+            webpage, 'embed', fatal=False)
+        data = re.sub('\\\\/', '/', data)
+        data2 = re.sub('\\\\/', '/', data2)
+        info = json.loads(data)
+        info2 = json.loads(data2)
 
-        data = self._download_json(
-            'http://videoarchiv.markiza.sk/json/video_jwplayer7.json',
-            video_id, query={'id': video_id})
-
-        info = self._parse_jwplayer_data(data, m3u8_id='hls', mpd_id='dash')
-
-        if info.get('_type') == 'playlist':
-            info.update({
-                'id': video_id,
-                'title': try_get(
-                    data, lambda x: x['details']['name'], compat_str),
-            })
-        else:
-            info['duration'] = parse_duration(
-                try_get(data, lambda x: x['details']['duration'], compat_str))
-        return info
+        formats = []
+        for format_id, format_list in info['tracks'].items():
+            if not isinstance(format_list, list):
+                format_list = [format_list]
+            for format_dict in format_list:
+                if not isinstance(format_dict, dict):
+                    continue
+                format_url = url_or_none(format_dict.get('src'))
+                format_type = format_dict.get('type')
+                ext = determine_ext(format_url)
+                if (format_type == 'application/x-mpegURL'
+                        or format_id == 'HLS' or ext == 'm3u8'):
+                    formats.extend(self._extract_m3u8_formats(
+                        format_url, video_id, 'mp4',
+                        entry_protocol='m3u8_native', m3u8_id='hls',
+                        fatal=False))
+                elif (format_type == 'application/dash+xml'
+                      or format_id == 'DASH' or ext == 'mpd'):
+                    formats.extend(self._extract_mpd_formats(
+                        format_url, video_id, mpd_id='dash', fatal=False))
+                else:
+                    formats.append({
+                        'url': format_url,
+                    })
+        thumbnail = info.get('plugins').get('thumbnails').get('url')
```

Comment:
Use `try_get(info, lambda x: x['plugins']['thumbnails']['url'], compat_str)` to avoid crashing on optional metadata.